core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -604,11 +603,11 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
         if (metrics != null)
           CoreUtils.swallow(metrics.close())
 
+        CoreUtils.swallow(AppInfoParser.unregisterAppInfo(jmxPrefix, config.brokerId.toString))
```

Comment:
Thinking about this some more, maybe it's safer to leave it as it was (with the `swallow` though). Not sure if any code is relying on the fact that the JMX registration is removed after the state and `isShutDown` is updated.